src/unix/fs.c

```diff
@@ -766,6 +770,102 @@ static ssize_t uv__fs_write(uv_fs_t* req) {
   return r;
 }
 
+static ssize_t uv__fs_copyfile(uv_fs_t* req) {
+#if defined(__APPLE__) && !TARGET_OS_IPHONE
+  /* On macOS, use the native copyfile(3). */
+  copyfile_flags_t flags;
+
+  flags = COPYFILE_ALL;
+
+  if (req->flags & UV_FS_COPYFILE_EXCL)
+    flags = flags | COPYFILE_EXCL;
+
+  return copyfile(req->path, req->new_path, NULL, flags);
+#else
+  uv_fs_t fs_req;
+  uv_file src;
+  uv_file dst;
+  struct stat statsbuf;
+  int dst_flags;
+  int result;
+  int err;
+
+  dst = -1;
+
+  /* Open the source file. */
+  src = uv_fs_open(req->loop, &fs_req, req->path, O_RDONLY, 0, NULL);
+  uv_fs_req_cleanup(&fs_req);
+
+  if (src < 0) {
+    err = src;
+    goto out;
+  }
+
+  /* Get the source file's mode. */
+  err = fstat(src, &statsbuf);
+
+  if (err != 0) {
```

Comment:
Just `if (fstat(...)) {`?  Assigning it to `err` hints it might be used further down when it's not.